### PR TITLE
Adds IPC only surgery - clone damage fix

### PR DESCRIPTION
**Pull request by EvilDragonfiend**
## About The Pull Request
* Adds IPC only surgery that fixes clone damage "repair prosthesis anomaly damage"

## Why It's Good For The Game
it is supposed to not happen generally, but zams sinful virology doomed their fate

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image

image

after trying this, I was able to fully fix IPC's clone damage

</details>

## Changelog
:cl:
add: IPC only new surgery "repair prosthesis anomaly damage" - can fix their clone damage
/:cl:

### CODE REVIEW

**Comment by bluezorua** [comment]
probably shouldn't need to use biological surgery tools (hemostat) to repair IPCs

**Comment by EvilDragonfiend** [comment]
> probably shouldn't need to use biological surgery tools (hemostat) to repair IPCs

sadly the real IPC surgery isn't that maintained, and hemostat is 'to pull out' something but I don't know which mechanic tool can be replaceable for it.

**Comment by lowwy** [comment]
> sadly the real IPC surgery isn't that maintained, and hemostat is 'to pull out' something but I don't know which mechanic tool can be replaceable for it.

Maybe an empty hand? I think one of the other surgeries on IPC also uses an empty hand.

**Comment by EvilDragonfiend** [comment]
> > sadly the real IPC surgery isn't that maintained, and hemostat is 'to pull out' something but I don't know which mechanic tool can be replaceable for it.
> 
> Maybe an empty hand? I think one of the other surgeries on IPC also uses an empty hand.

since surgery steps are a bit shitcode, it requires more refactoring if it should need hands.

**Comment by Penwin0** [comment]
> It is supposed to not happen generally, but zams sinful virology doomed their fate

I'd rather that instead of adding a very specific surgery not really for a snowflake type of damage they aren't supposed to take in the first place that whatever viro symptom causes this to stop doing so on IPCs

**Comment by Tsar-Salat** [comment]
> but I don't know which mechanic tool can be replaceable for it

Crowbar. You leverage stuff out with it. (Tiles, glass screens on consoles, etc.)

**Comment by EvilDragonfiend** [comment]
> I'd rather that instead of adding a very specific surgery not really for a snowflake type of damage they aren't supposed to take in the first place that whatever viro symptom causes this to stop doing so on IPCs

well, maybe it might be enough reason to close this PR since they shouldn't play such weird behavior...

but for me, IPC as cell damage is interpreted as 'atom integrity & stability damage'. so I believe IPC having this damage isn't really 'not makes sense'

**Comment by EvilDragonfiend** [comment]
okay, what if IPC takes 3x brute damage from cell damage? 1 cell damage will be replaced as 3 brute damage to IPC.

File: code/modules/surgery/ipc_surgery.dm [hunk 1]
```diff
@@ -0,0 +1,68 @@
+/datum/surgery/ipc_clone_damage_surgery
+	name = "repair prosthesis anomaly damage"
```
Comment (PowerfulBacon) [inline code/modules/surgery/ipc_surgery.dm:2]: Capitalise this like the other surgeries

File: code/modules/surgery/ipc_surgery.dm [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+		/datum/surgery_step/cut_wires, // 6
+		/datum/surgery_step/prepare_electronics/ipc_clone_damage_surgery, // 7
+		/datum/surgery_step/pulling_out, // 8
+		/datum/surgery_step/insert_wires, // 9
+		/datum/surgery_step/prepare_electronics/ipc_clone_damage_surgery, // 10
+		/datum/surgery_step/mechanic_open/ipc_clone_damage_surgery, // 11
+		/datum/surgery_step/prepare_electronics/ipc_clone_damage_surgery, // 12
+		/datum/surgery_step/mechanic_wrench/ipc_clone_damage_surgery, // 13
+		/datum/surgery_step/mechanic_wrench, // 14
+		/datum/surgery_step/mechanic_close // 15
```
Comment (PowerfulBacon) [inline code/modules/surgery/ipc_surgery.dm:23]: This is an insane amount of steps

File: code/modules/surgery/ipc_surgery.dm [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+// bolt step
+/datum/surgery_step/mechanic_open/ipc_clone_damage_surgery
+	name = "retighten small bolts"
+	time = 40
+
+/datum/surgery_step/mechanic_open/ipc_clone_damage_surgery/preop(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
+	display_results(user, target, "<span class='notice'>You begin to adjust small bolts in [target]'s [parse_zone(target_zone)]...</span>",
+		"[user] begins to adjust small bolts in [target]'s [parse_zone(target_zone)].",
+		"[user] begins to adjust small bolts in [target]'s [parse_zone(target_zone)].")
```
Comment (PowerfulBacon) [inline code/modules/surgery/ipc_surgery.dm:46]: Should these have spans?

File: code/modules/surgery/ipc_surgery.dm [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+// wrench step - this will be the final step
+/datum/surgery_step/mechanic_wrench/ipc_clone_damage_surgery
+	name = "retighten big bolts"
+	time = 80
+
+/datum/surgery_step/mechanic_wrench/ipc_clone_damage_surgery/preop(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
+	display_results(user, target, "<span class='notice'>You begin to carefully calibrate functional bolts in [target]'s [parse_zone(target_zone)]...</span>",
+		"[user] begins to carefully calibrate functional bolts in [target]'s [parse_zone(target_zone)].",
+		"[user] begins to carefully calibrate functional bolts in [target]'s [parse_zone(target_zone)].")
```
Comment (PowerfulBacon) [inline code/modules/surgery/ipc_surgery.dm:61]: Here too

File: code/modules/surgery/mechanic_steps.dm [hunk 5]
```diff
@@ -103,3 +103,62 @@
...
+			"[user] begins to insert some wires in [target]'s [parse_zone(target_zone)].")
+
+/datum/surgery_step/insert_wires/tool_check(mob/user, obj/item/stack/cable_coil/tool)
+	if(tool.amount < 10)
+		to_chat(user, "<span class='warning'>You need to have 10 pieces of wires to operate this.</span>")
+		return FALSE
+	if(tool.usesound)
+		preop_sound = tool.usesound
+
+	tool.amount -= 10
```
Comment (PowerfulBacon) [inline code/modules/surgery/mechanic_steps.dm:151]: This should use use() rather than subtracting amount, so you don't result with 0 amount stacks

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
Should we fix IPCs getting clone damage in the first place?

**Comment by EvilDragonfiend** [comment]
> Should we fix IPCs getting clone damage in the first place?

Yeah, that's questionable since only it's virology fuckery. So, I am thinking of dealing 3x brute damage conversion per clone damage in case they take.

**Pull request closed without merging**